File: python3/2381.py
class Solution:
    def shiftingLetters(self, s: str, sh: List[List[int]]) -> str:
        alpha = "abcdefghijklmnopqrstuvwxyz"
        n1 = len(s)
        ch = [0] * (n1 + 1)
        for i, j, k in sh:
            ch[i] += k * 2 - 1
            ch[j + 1] += 1 - k * 2
        ps = 0
        c = []
        for i in range(n1):
            ps += ch[i]
            c.append(alpha[(alpha.find(s[i]) + ps) % 26])
        return "".join(c)

File: javascript/2381.js
/**
 * @param {string} c
 * @param {number[][]} sh
 * @return {string}
 */
var shiftingLetters = function (c, sh) {
    const alpha = "abcdefghijklmnopqrstuvwxyz";
    let n1 = c.length, n2 = sh.length, ps = 0;
    const ch = new Array(n1 + 1).fill(0);
    for (let i of sh) {
        ch[i[0]] += i[2] * 2 - 1;
        ch[i[1] + 1] += -(i[2] * 2 - 1);
    }
    const res = [];
    for (let i = 0; i < n1; i++) {
        ps += ch[i];
        res.push(alpha[(alpha.indexOf(c[i]) + (ps % 26) + 26) % 26]);
    }
    return res.join("");
};
